MSGraphClientSDK/MSGraphClientSDK/GraphTasks/MSLargeFileUploadTask.m

```diff
@@ -0,0 +1,98 @@
+//
+// Copyright (c) Microsoft Corporation. All Rights Reserved. Licensed under the MIT License. See License in the project root for license information.
+//
+
+#import "MSLargeFileUploadTask.h"
+#import "MSURLSessionDataTask.h"
+
+@interface MSLargeFileUploadTask()
+
+@property (nonatomic, strong) MSHTTPClient *httpClient;
+@property (nonatomic, strong) NSData *fileData;
+@property (nonatomic, strong) NSDictionary *uploadSessionDictionary;
+@property (nonatomic) NSInteger chunkSize;
+@property (nonatomic) NSRange currentRange;
+
+@end
+
+@implementation MSLargeFileUploadTask
+
+-(instancetype)initWithClient:(MSHTTPClient *)httpClient fileData:(NSData *)fileData uploadSessionDictionary:(NSDictionary *)uploadSessionDictionary andChunkSize:(NSInteger)chunkSize
+{
+    self = [super init];
+    if(self)
+    {
+        self.httpClient = httpClient;
+        NSAssert(self.httpClient, @"HTTP Client is requried to make API calls");
```

Comment:
Logging these values are required ?

Reply:
Its not a log, it's an Assert. In case there is not httpClient passed, this will stop the execution as httpClient is required for execution of upload task.